lib/core/token.js

```diff
@@ -0,0 +1,246 @@
+import { addTokenPrefix } from './utilities'
+import jwtDecode, { InvalidTokenError } from 'jwt-decode'
+
+export class Token {
+  constructor (auth) {
+    this.$auth = auth
+  }
+
+  _setHeader (token) {
+    if (this.$auth.strategy.options.token.global) {
+      // Set Authorization token for all axios requests
+      this.$auth.ctx.app.$axios.setHeader(this.$auth.strategy.options.token.name, token)
```

Comment:
In continuation of custom request handler PR, what do you think to refactor this into an abstract `RequestHandler` class?

Reply:
Great idea! 😄

Reply:
Refactored in commit https://github.com/nuxt-community/auth-module/pull/548/commits/db3ecea761b421500b0dbfdb38727d5153f944f4